browser/resources/settings/brave_rewards_page/brave_rewards_browser_proxy.js

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {addSingletonGetter} from 'chrome://resources/js/cr.m.js';
+
+/** @interface */
+export class BraveRewardsBrowserProxy {
+  isRewardsEnabled() {}
+  getRewardsParameters() {}
+}
+
+/**
+ * @implements {settings.BraveRewardsBrowserProxy}
+ */
+export class BraveRewardsBrowserProxyImpl {
+  /** @override */
+  isRewardsEnabled () {
+    return new Promise((resolve) => chrome.braveRewards.shouldShowOnboarding(
```

Comment:
FYI: there's also a `chrome.braveRewards.getRewardsEnabled` function for just getting that pref (if you even need that from the settings page).

I would like to see us deprecate `shouldShowOnboarding` (and certainly not use it here), because its usage is not well-defined. It looks like for this use case we *should* just be able to use the rewards.enabled pref directly (since it just controls whether the "params" are downloaded)?